test/testdrive/kafka-upsert-debezium-sources.td

```diff
@@ -62,7 +62,7 @@ $ kafka-ingest format=avro topic=dbzupsert key-format=avro key-schema=${keyschem
   ENVELOPE DEBEZIUM
 
 ! SELECT * FROM start_offset_misconfigured
-Updates and deletes are not allowed for this source! This probably means it was started with `start_offset` and without `UPSERT` semantics.
+Invalid data in source, saw retractions (1) for row that does not exist
```

Comment:
the old message seems more informative

Reply:
The previous code was unconditionally erroring out if you passed `start_offset`, meaning that even if a user had a legitimate reason to start reading the topic from a later point (perhaps there are garbage in the beginning), they couldn't.

This doesn't sound right so this PR makes it so that we will always attempt to process what the user instructed us to do. If it turns out that the data is non-sensical an error will be produced, otherwise we'll produce a collection